vendor: No-change bump to pebble

itsbilal opened the pull request:
This bump lets us pick up some iterator fixes in pebble, some
RocksDB manifest compatibility fixes, as well as debug tool
checks.

All pebble changes picked up:
https://github.com/cockroachdb/pebble/compare/efdb7f6d67b47189a442e2235a57bf9ae9c25bdb...1acac2061f9f5430a7c69e1e5afedfdaeab9e4ef

Release note: None

Release justification: Improves debugging capability, low-risk
because pebble is currently only used to write to sstables.

Commit:
vendor: No-change bump to pebble

This bump lets us pick up some iterator fixes in pebble, some
RocksDB manifest compatibility fixes, as well as debug tool
checks.

All pebble changes picked up:
https://github.com/cockroachdb/pebble/compare/efdb7f6d67b47189a442e2235a57bf9ae9c25bdb...1acac2061f9f5430a7c69e1e5afedfdaeab9e4ef

Release note: None

Release justification: Improves debugging capability, low-risk
because pebble is currently only used to write to sstables.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/41080)
<!-- Reviewable:end -->

petermattis commented:
Hmm, Pebble is complaining about a background error that "should never happen". `pebble: flush next log number is unset`. Let me track down what is going on there.

petermattis commented:
Well, that error is super reproducible using `BenchmarkPebbleMapWrite`.

petermattis commented:
Ah, the problem is that we've set `DisableWAL` and the error check isn't accounting for that. Phew! I thought this was something serious.

itsbilal commented:
Thanks for fixing that quickly @petermattis ! I've bumped pebble again to include it.

petermattis reviewed (approved):
LGTM, though I think you need to update the commit message to include the new end commit.

itsbilal commented:
Bumped pebble again to pick up commits from yesterday, updated the PR description and commit message to specify the new commit range, and rebased vendor to fix a conflict there. Thanks for the review!

itsbilal commented:
Build passed, I'll merge this. Thanks!

bors r+
